lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
So you are suggesting do not use GC.Collect....Regarding reuse - its really
a tradeoff between memory and speed. There are going to be memory overheads
due to pipelining and all if u want speed. So I am not sure what can be
done over short term. Long term ofcourse we should try to fix it.

I will suggest going with this short term solution for now for first demo
and version of TLC++.

On Tue, Sep 29, 2015 at 1:03 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40721524>:
>
> >              {
> > +                GC.Collect();
>
> Regarding the .NET version: I believe we currently *target* 4.5. That is
> independent of what you have on your machine.
>
> Also, it seems we are really dealing with two issues:
>
>    1. IMRU uses too much memory and is inefficient in its object
>    (de-)allocations. Some object reuse as suggested by @anupam128
>    <https://github.com/anupam128> should help.
>    2. The memory constraint of a container isn't enforced in a way that
>    the CLR knows about. This means that GC isn't invoked when it should.
>    Calling the GC on our end won't really solve that issue, just delay it.
>
> Can we focus on the 1st problem in this issue / PR and create a JIRA for
> the second one?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40721524>.
>